File: LeetCode/Easy/0219-contains-duplicate-ii/0219-contains-duplicate-ii.cs
public class Solution {
    public bool ContainsNearbyDuplicate(int[] nums, int k) {
        var set = new HashSet<int>();
        for (var i = 0; i < nums.Length; i++)
        {
            if (i > k)
            {
                set.Remove(nums[i - k - 1]);
            }
            if (!set.Add(nums[i]))
            {
                return true;
            }

        }

        return false;
    }

    // alternative
    public bool ContainsNearbyDuplicateDic(int[] nums, int k) {
        var dic = new Dictionary<int, int>();
        for (var i = 0; i < nums.Length; i++)
        {
            if (dic.TryGetValue(nums[i], out int lastIndex) && i - lastIndex <= k)
            {
                return true;
            }
            dic[nums[i]] = i;
        }

        return false;
    }
}

File: LeetCode/Easy/0219-contains-duplicate-ii/0219-contains-duplicate-ii.py
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        num_2_position = {}
        for i, n in enumerate(nums):
            if n in num_2_position and (i - num_2_position[n]) <= k:
                return True
            num_2_position[n] = i

        return False